lib/portlayer/event/collector/vsphere/collector.go

```diff
@@ -171,15 +181,20 @@ func (ec *EventCollector) Start() error {
 		// the manager will be closed with the session
 
 		for {
-			err := ec.vmwManager.Events(ctx, refs, pageSize, followStream, force, func(_ types.ManagedObjectReference, page []types.BaseEvent) error {
+			err := ec.vmwManager.Events(controlContext, refs, pageSize, followStream, force, func(_ types.ManagedObjectReference, page []types.BaseEvent) error {
 				evented(ec, page)
 				return nil
 			}, eventTypes...)
 			// TODO: this will disappear in the ether
 			if err != nil {
 				log.Debugf("Error configuring %s: %s", name, err.Error())
 			}
+
+			// this is a VERY basic throttle so that we don't DoS the remote when the client is NotAuthenticated.
+			// this should be removed/replaced once there is structured connection management in place to trigger re-authentication as required.
+			time.Sleep(2 * time.Second)
```

Comment:
Do I recall you mentioning that we had a standardized way to do back-off?

Reply:
`retry` package, there are some good examples around in the code.  see https://github.com/vmware/vic/blob/master/pkg/retry/retry.go

Reply:
We do have the retry but we have knowledge about the keepalive triggering reauth (20s at time time) and don't expect to need or want exponential backoff. That package also assumes time-bounds (e.g. Max retry time) whereas this is expected to be indefinite.
Basically while I could probably force the backoff package to do basic throttling I don't think it's a good fit here. We may want an actual throttle package, however what we're really after here structured connection management so this doesn't have to occur inline in product code at all.